### PR TITLE
Let umask sort out mirror conf dir perms

### PR DESCRIPTION
**Pull request by bwarden**
Set /etc/swupd permissions to 0777 to let the default umask policy
handle stripping bits appropriately.

### CODE REVIEW

File: src/mirror.c [hunk 1]
```diff
@@ -74,13 +74,10 @@ static int write_to_path(char *content, char *path)
...
-	int ret = mkdir(dir, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
-	char *cmd;
-	string_or_die(&cmd, "mkdir -p %s", dir);
-	ret = system(cmd);
-	free_string(&cmd);
+	int ret = mkdir(dir, S_IRWXU | S_IRWXG | S_IRWXO);
 	free_string(&tmp);
-	if (ret) {
+
+	if (ret && EEXIST != errno) {
```
Comment (bwarden) [inline src/mirror.c]: May need to expand on this. Line 128 in the error path (unset_mirror_url()) means that a pre-existing _file_ /etc/swupd would be unlinked, although maybe that's not necessarily fatal.

**Comment by matthewrsj** [comment]
@icarus-sparry the man pages test is failing, even though he hasn't updated the man pages?

**Comment by icarus-sparry** [comment]
Hmm. So the test shows that the 'make' command ran rst2man to update swupd.1, and then `git  diff-index --quiet` showed that the checked in file differed from the one in docs.

Maybe I can't write tests?

Maybe docutils has been updated so rst2man produces different output?

**Comment by matthewrsj** [comment]
Re-running travis seems to make the test pass...

**Comment by bwarden** [comment]
I'm working through some corner cases for pre-existence of /etc/swupd. Once I have the tests implemented and passing, I'll update this PR.

File: src/mirror.c [hunk 2]
```diff
@@ -208,6 +232,13 @@ static int parse_options(int argc, char **argv)
 			}
 			unset = true;
 			break;
+		case 'F':
```
Comment (matthewrsj) [inline src/mirror.c]: Why in the world is this necessary for setting a mirror URL?

Comment (bwarden) [reply src/mirror.c]: Without -F staging, swupd mirror fails in the test environment.

Comment (matthewrsj) [reply src/mirror.c]: Then let's fix the test environment instead of adding a useless command to the interface.

Comment (matthewrsj) [reply src/mirror.c]: In your test setup do this:
```bash
sudo mkdir -p "$DIR/target-dir/usr/share/defaults/swupd"
echo 1 | sudo tee "$DIR/target-dir/usr/share/defaults/swupd/format"
```

Comment (bwarden) [reply src/mirror.c]: Do I just need to throw an arbitrary positive integer into "$DIR/target-dir/usr/share/defaults/swupd/format"?

Comment (matthewrsj) [reply src/mirror.c]: yes

**Comment by bwarden** [comment]
Cleaned up and squashed

File: test/functional/mirror/createdir-negative/test.bats [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+teardown() {
+  sudo rm -rf "$DIR/target-dir/etc/swupd"
+  sudo rm -rf "$DIR/target-dir/foo"
+}
+
+@test "mirror /etc/swupd is a file" {
+  sudo rm -rf "$DIR/target-dir/etc/swupd"
+  sudo touch "$DIR/target-dir/etc/swupd"
+  [[ -f "$DIR/target-dir/etc/swupd" ]] && ! [[ -L "$DIR/target-dir/etc/swupd" ]]
+  run sudo sh -c "$SWUPD mirror -s http://example.com/swupd-file $SWUPD_OPTS_MIRROR"
```
Comment (matthewrsj) [inline test/functional/mirror/createdir-negative/test.bats]: Can you add a `[ $status != 0 ]` check here as well? And in the other test.

File: test/functional/mirror/createdir/test.bats [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+teardown() {
+  sudo rm -rf "$DIR/target-dir/etc/swupd"
+  sudo rm -rf "$DIR/target-dir/foo"
+}
+
+@test "mirror /etc/swupd does not exist" {
+  sudo rm -rf "$DIR/target-dir/etc/swupd"
+  run sudo sh -c "$SWUPD mirror -s http://example.com/swupd-file $SWUPD_OPTS_MIRROR"
```
Comment (matthewrsj) [inline test/functional/mirror/createdir/test.bats]: Please check `[ $status -eq 0 ]`

File: test/functional/mirror/createdir/test.bats [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  check_lines "$output"
+  [[ "http://example.com/swupd-file" == $(<$DIR/target-dir/etc/swupd/mirror_contenturl) ]]
+  [[ "http://example.com/swupd-file" == $(<$DIR/target-dir/etc/swupd/mirror_versionurl) ]]
+}
+
+@test "mirror /etc/swupd already exists" {
+  sudo mkdir "$DIR/target-dir/etc/swupd"
+  run sudo sh -c "$SWUPD mirror -s http://example.com/swupd-file $SWUPD_OPTS_MIRROR"
```
Comment (matthewrsj) [inline test/functional/mirror/createdir/test.bats]: Please check `[ $status -eq 0 ]`

File: test/functional/mirror/createdir/test.bats [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  [[ "http://example.com/swupd-file" == $(<$DIR/target-dir/etc/swupd/mirror_contenturl) ]]
+  [[ "http://example.com/swupd-file" == $(<$DIR/target-dir/etc/swupd/mirror_versionurl) ]]
+}
+
+@test "mirror /etc/swupd is a symlink to a directory" {
+  sudo rm -rf "$DIR/target-dir/etc/swupd"
+  sudo rm -rf "$DIR/target-dir/foo"
+  sudo mkdir "$DIR/target-dir/foo"
+  sudo ln -s "$DIR/target-dir/foo" "$DIR/target-dir/etc/swupd"
+  run sudo sh -c "$SWUPD mirror -s http://example.com/swupd-file $SWUPD_OPTS_MIRROR"
```
Comment (matthewrsj) [inline test/functional/mirror/createdir/test.bats]: Please check `[ $status -eq 0 ]`
